src/utils/utils.cpp

```diff
@@ -161,6 +162,25 @@ QDir mp::utils::base_dir(const QString& path)
     return info.absoluteDir();
 }
 
+std::filesystem::path mp::utils::normalize_path(const std::filesystem::path& path)
```

Comment:
Since we're replacing Qt's [QDir::cleanPath](https://github.com/qt/qtbase/blob/26c45a1b534752df4ebd89206079cd7cfe55df84/src/corelib/io/qdir.cpp#L2517), we need to ensure that this function behaves as intended. Can we have unit tests to ensure that? QT's own test case for testing the cleanPath might be a good starting point:

https://github.com/qt/qtbase/blob/26c45a1b534752df4ebd89206079cd7cfe55df84/tests/auto/corelib/io/qdir/tst_qdir.cpp#L1328